### ISSUE TITLE
[ENG-225] Integration with environment secrets in GitHub

### ISSUE THREAD

**Issue by vmatsiiako** [Infisical/infisical#1154]
### Feature description
Right now, Infisical integrated only with repository secrets. It would be great to also integrate with environment secrets. 

### Why would it be useful?
This integration has been requested by many users. 

<sub>[ENG-225](https://linear.app/infisical/issue/ENG-225/integration-with-environment-secrets-in-github)</sub>

**Comment by kisharnath** [comment]
Hey can I work on this?

**Comment by vmatsiiako** [comment]
Hey @kisharnath! Absolutely! Please let us know if you need any help. You should also consider joining our community for contributors on Slack: https://infisical.com/slack

**Comment by kisharnath** [comment]
Hey I am trying to understand the problem , Can you elaborate the problem more ?

**Comment by vmatsiiako** [comment]
Infisical is currently able to sync to GitHub repository secrets. Ideally, we also want an option to sync to environment secrets. Does that make sense?

![image](https://github.com/Infisical/infisical/assets/78047717/b128ad14-368a-43f3-8ff6-749fac8125c0)

**Comment by vmatsiiako** [comment]
Hey @kisharnath! Just wanted to see if you are still working on this issue? Otherwise I'm happy to assign it to someone else. Just let me know!

**Comment by kisharnath** [comment]
I am working on this issue I will send PR in this week

**Comment by kisharnath** [comment]
Hi vlad , I have implemented the code but I am facing the problem of syncing secret to repo in local developement , Could you please help me in this.Even the repository secret are not syncing

**Comment by Shraeyas** [comment]
@vmatsiiako Is this still open? I can have a look at this if there's no active work going on.

**Comment by vmatsiiako** [comment]
Hi @kisharnath! Would you be able to ask this in the #contributing channel of our Slack community? (https://infisical.com/slack)

**Comment by Shraeyas** [comment]
@vmatsiiako If this is not under active dev I would love to have a look at this issue.

**Comment by vmatsiiako** [comment]
@Shraeyas yes, sounds good, that's good. @kisharnath please let us know if you are still working on this or not.

**Comment by kisharnath** [comment]
I am very very sorry for this delay. But I was having some health issues now I am fine I can work on this .

**Comment by Salman2301** [comment]
#1398
